src/modules/Board/components/TaskFormDialog.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  Button,
+  Stack,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  TextField,
+  FormControl,
+  InputLabel,
+  MenuItem,
+} from '@mui/material';
+import Select from '@mui/material/Select';
+import { useEffect, useCallback } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { Dialog as DialogType } from '@/types/dialog';
+import { Status } from '@/types/status';
+import { Task } from '@/types/task';
+import { User } from '@/types/user';
+
+type TaskDialogType = {
+  dialog: DialogType;
+  isCreate: boolean;
+  currentTask?: Task;
+  users: User[];
+  statuses: Status[];
+};
+
+type TaskForm = {
+  name: string;
+  description: string;
+  userId: string;
+  status: string;
+};
+
+const TASK_FORM_DEFAULT_VALUES: TaskForm = {
+  name: '',
+  description: '',
+  userId: '',
+  status: 'status1',
+};
+
+export function TaskFormDialog({
+  dialog,
+  isCreate,
+  currentTask,
+  users,
+  statuses,
+}: TaskDialogType) {
+  const { control, setValue } = useForm({
+    defaultValues: TASK_FORM_DEFAULT_VALUES,
+  });
+
+  useEffect(() => {
+    if (!isCreate && currentTask) {
+      setValue('name', currentTask.name);
+      setValue('description', currentTask.description);
+      setValue('userId', currentTask.assignedTo);
+      setValue('status', currentTask.statusId);
+    } else {
+      setValue('name', '');
+      setValue('description', '');
+      setValue('userId', '');
+      setValue('status', 'status1');
+    }
+  }, [isCreate, setValue, currentTask]);
```

Comment:
I think it can be better
```suggestion
  const { control, reset } = useForm({
    defaultValues: TASK_FORM_DEFAULT_VALUES,
  });

  useEffect(() => {
  const initialValues = !isCreate && currentTask ? currentTask : TASK_FORM_DEFAULT_VALUES;
  reset(initialValues);
  }, [isCreate, reset, currentTask]);
```